pkg/stub/handler.go

```diff
@@ -327,6 +337,75 @@ func (h *Handler) updateCfgArch(cfg *v1.Config) *v1.Config {
 	return cfg
 }
 
+func (h *Handler) imageConfigBlocksImageStreamCreation(name string) bool {
+	//TODO openshift/client-go/config/clientset/versioned/fake and ConfigV1Interface has compile issues with
+	// respect to ConfigV1Client (missing method implementations), so for now we cannot use it in our
+	// unit tests.  In the actual runtime if we cannot create the client the operator errors out on startup.
+	if h.configclient == nil {
+		return false
+	}
+	err := wait.PollImmediate(5*time.Second, 20*time.Second, func() (done bool, err error) {
+		// if the image config allowed registry or blocked registry list will prevent the creation of imagestreams,
+		// we consider this inaccessible
+		imgCfg, err := h.configclient.Images().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			logrus.Printf("unable to retrieve image configuration as part of testing registry.redhat.io connectivity: %s", err.Error())
+			return false, nil
+		} else {
+			// check allowed domain list
+			if len(imgCfg.Spec.AllowedRegistriesForImport) > 0 {
+				for _, rl := range imgCfg.Spec.AllowedRegistriesForImport {
+					logrus.Printf("considering allowed registries domain %s for %s", rl.DomainName, name)
+					if strings.HasSuffix(name, rl.DomainName) {
+						msg := fmt.Sprintf("the allowed registries domain %s allows imagestream creation access for search param %s", rl.DomainName, name)
+						logrus.Printf(msg)
```

Comment:
```suggestion
						logrus.Printf(("the allowed registries domain %s allows imagestream creation access for search param %s", rl.DomainName, name)
```

Reply:
This could be updated throughout the pull request. There shouldn't be a reason to create the `msg` variable and then pass it into `Printf` or `Errorf` unless it is going to be used again somewhere else.

Reply:
yeah there were some cases where I was both creating a msg for a log as well as for an error to return

in those cases I'll try to address

in the cases where both are in fact occurring, I'll keep the pattern of defining the string once